src/chapters/http/http-responses.rst

```diff
@@ -0,0 +1,163 @@
+Responses
+=========
+
+.. index::
+   single: HTTP; response
+
+Every time a web server receives an HTTP request, it sends an HTTP response
+back to the client.
+
+A generic HTTP response looks like this:
+
+::
+
+   HTTP/2.0 200 OK
+   Date: Wed, 22 May 2020 17:36:50 GMT
+   Content-Type: text/html; charset=UTF-8
+   Content-Length: 8050
+   Last-Modified: Wed, 22 May 2020 17:33:45 GMT
+
+   <!DOCTYPE html>
+   <html>
+   <head>
+   <meta charset="UTF-8">
+   <!--Rest of HTML page -->
+   </html>
+
+The structure has these components:
+
+.. index::
+   single: HTTP; response code
+   single: HTTP; response headers
+   single: HTTP; response line
+
+#. **Response line:** The first line of the response contains status
+   information like the **response code**. In this example, the response code
+   is ``200``, which indicates the request was successfully fulfilled.
+#. **Response headers:** Below the response line are the **response headers**.
+   Similar to request headers, these are key-value pairs that contain data
+   about the response. In the example above, these include information about
+   the date and time, the data being sent (text and HTML), the size of the
+   response body, and when it was last modified.
+#. **Blank line:** This marks the end of the response headers.
+#. **Response body (Optional):** Below the blank line, the request body takes
+   up the rest of the HTTP response. This usually includes the HTML and CSS for
+   the requested webpage.
+
+Response Codes
+--------------
+
+We've all gotten a ``404 File Not Found`` error code at some point when we've
+surfed the web.
+
+.. figure:: figures/404.png
+   :alt: A 404 message from GitHub displayed in a browser.
+   :width: 80%
+
+   Did we make a mistake when typing the URL?
+
+While ``404`` is the most common error we see when clicking links or
+typing in URLs, other codes exist as well.
+
+.. index::
+   single: HTTP; response code
+
+Servers use HTTP **response codes** to report the result when they try to
+answer a client's request. The codes are always three-digit numbers, and they
+fall into one of five categories:
+
+#. **Information**: (1xx) The request was received, but the server is still
+   working on the response.
+#. **Success**: (2xx) The request was valid and the server successfully
+   responded.
+#. **Redirect**: (3xx) The request is incomplete. The client must take
+   additional action before the server can respond.
```

Comment:
I like this a bit better, with more context: "The client must take additional action, such as send the request to a different server, before the server can respond."

Reply:
Fixed.